clustersConfig.py

```diff
@@ -226,16 +228,25 @@ def __init__(
         self.configured_workers = [NodeConfig(self.name, **w) for w in cc["workers"]]
         self.workers = [NodeConfig(self.name, **w) for w in worker_range.filter(cc["workers"])]
 
-        if self.kind == "openshift":
-            self.configure_ip_range()
-
         self.set_cc_hosts_defaults(cc)
+
+        for e in cc["hosts"]:
+            self.hosts.append(HostConfig(**e))
+
+        if test_only:
```

Comment:
I believe that the fact that we have tests should not show up as booleans throughout the code. Can we rework this?

Reply:
These are the first patches from a larger set that rework `clustersConfig.py`. It adds the tests first, before doing rework. I'd prefer if tests can be merged first, before changing the code (and not the other way around).

As the commit message says, this will change later.

Reply:
So the new function will still follow the following structure?

```
def func():
    # part 1 without side effects

    # if somebool:
    #   part 2 with side effects?
```